jlm/rvsdg/simple-node.hpp

```diff
@@ -120,6 +134,31 @@ SimpleNode::output(size_t index) const noexcept
   return static_cast<simple_output *>(Node::output(index));
 }
 
+struct CreateOpNodeResult
+{
+  Node * node;
+  std::vector<output *> outputs;
+};
+
+template<typename OpType, typename... OpParameters>
+CreateOpNodeResult
+CreateOpNode(const std::vector<output *> & arguments, OpParameters... opParameters)
+{
+  Region * region = arguments.at(0)->region();
+  for (const auto & argument : arguments)
+  {
+    JLM_ASSERT(argument->region() == region);
+  }
+  Node * node =
+      SimpleNode::Create(region, std::make_unique<OpType>(std::move(opParameters)...), arguments);
+  std::vector<output *> outputs;
+  for (std::size_t n = 0; n < node->noutputs(); ++n)
+  {
+    outputs.push_back(node->output(n));
+  }
```

Comment:
We have the outptus() function that does exactly this: https://github.com/phate/jlm/blob/6db940a6220bd6be1ba2203edceb6269114ea4b6/jlm/rvsdg/node.hpp#L1052C1-L1059C2

Reply:
obsolete since returning node now